cpp/include/cuvs/neighbors/ivf_pq.h

```diff
@@ -334,6 +361,58 @@ cuvsError_t cuvsIvfPqBuild(cuvsResources_t res,
                            cuvsIvfPqIndexParams_t params,
                            DLManagedTensor* dataset,
                            cuvsIvfPqIndex_t index);
+
+/**
+ * @brief Build a IVF-PQ index from existing centroids and codebook. The input tensors must be on
+ *        device memory.
+ *
+ * @code {.c}
+ * #include <cuvs/core/c_api.h>
+ * #include <cuvs/neighbors/ivf_pq.h>
+ *
+ * // Create cuvsResources_t
+ * cuvsResources_t res;
+ * cuvsError_t res_create_status = cuvsResourcesCreate(&res);
+ *
+ * // Assume populated `DLManagedTensor` types here
+ * DLManagedTensor pq_centers;
+ * DLManagedTensor centers;
+ * DLManagedTensor rotation_matrix;
+ *
+ * // Create default index params
+ * cuvsIvfPqIndexParams_t index_params;
+ * cuvsError_t params_create_status = cuvsIvfPqIndexParamsCreate(&index_params);
+ *
+ * // Create IVF-PQ index
+ * cuvsIvfPqIndex_t index;
+ * cuvsError_t index_create_status = cuvsIvfPqIndexCreate(&index);
+ *
+ * // Build the IVF-PQ Index
+ * cuvsError_t build_status = cuvsIvfPqBuildFromCentroids(res, index_params, dim, &pq_centers,
+ *   &centers, &rotation_matrix, index);
+ *
+ * // de-allocate `index_params`, `index` and `res`
+ * cuvsError_t params_destroy_status = cuvsIvfPqIndexParamsDestroy(index_params);
+ * cuvsError_t index_destroy_status = cuvsIvfPqIndexDestroy(index);
+ * cuvsError_t res_destroy_status = cuvsResourcesDestroy(res);
+ * @endcode
+ *
+ * @param[in] res cuvsResources_t opaque C handle
+ * @param[in] params cuvsIvfPqIndexParams_t used to build IVF-PQ index
+ * @param[in] dim dimensionality of the input data
```

Comment:
Seems like we should know this from the pqindex, right? At least, I can't see why this wouldn't be stored on the index, especially since it's responsible for storing the lists.